hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/spark/sql/hudi/TestAlterTableDropPartition.scala

```diff
@@ -399,7 +399,9 @@ class TestAlterTableDropPartition extends HoodieSparkSqlTestBase {
         // drop 2021-10-01 partition
         spark.sql(s"alter table $tableName drop partition (year='2021', month='10', day='01')")
 
-        spark.sql(s"""insert into $tableName values (2, "l4", "v1", "2021", "10", "02")""")
+        withSQLConf("hoodie.datasource.write.operation" -> "upsert") {
```

Comment:
can we add comments on why we had to do this? I  am assuming you are explicitly setting the op type to "upsert" only where it makes sense and not just to make the test case pass. Just wanted to ensure we don't unintentionally hide any bugs by doing this

Reply:
This test I'm not sure about. The others in this file make sense to not use bulk insert because they need log files